src/main/scala/org/ergoplatform/ErgoBox.scala

```diff
@@ -93,7 +94,7 @@ object ErgoBox {
     val size: Short = 32
   }
 
-  type TokenId = Digest32
```

Comment:
Digest32 type was introduced for a reason introduce distinction between atbitrary-sized byte arrays and 32-bytes arrays. Not clear why new type is introduced.

Reply:
1) This is to optimize org.ergoplatform.ErgoLikeContext.toSigmaContext which takes about 32% of times mentioned here ergoplatform/ergo#645
2) Digest32 is not true new type, it is still mutable array without correct equals and hashcode. With Coll it will be more safe and also more efficient by implementing special hashCode with will use first 4 bytes from the digest